python/mlx/_os_warning.py

```diff
@@ -0,0 +1,6 @@
+import platform
+
+if platform.system() == "Darwin":
+    major, minor, _ = map(int, platform.mac_ver()[0].split("."))
+    if (major, minor) < (13, 5):
+        raise OSError(f"Only macOS 13.5 and newer are supported, not {major}.{minor}")
```

Comment:
Is `OSError` the right one here? I would use `ImportError`. Either way it's fine.